sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1452,6 +1463,27 @@ def _snapshot_to_json(snapshot: Snapshot) -> str:
     return snapshot.json(exclude={"intervals", "dev_intervals"})
 
 
+def parse_snapshot(
+    model_cache: ModelCache,
+    serialized_snapshot: str,
+    name: str,
+    identifier: str,
+    seed_content: t.Optional[str],
+) -> Snapshot:
+    payload = json.loads(serialized_snapshot)
+
+    def loader() -> Node:
+        return parse_obj_as(Node, payload["node"])  # type: ignore
+
+    payload["node"] = model_cache.get_or_load(f"{name}_{identifier}", loader=loader)  # type: ignore
+    snapshot = Snapshot(**payload)
+
+    if seed_content and isinstance(snapshot.node, SeedModel):
+        snapshot.node = t.cast(SeedModel, snapshot.node).to_hydrated(seed_content)
```

Comment:
why did you have to cast, there's an isinstance check on 1481

Reply:
seed_content could be None, is that ok? the previous code checked that seed_content was not falsey

Reply:
Removed cast. The `if seed_content` does this check.